### PR TITLE
New Option class to handle values that might not be present 

### PR DESCRIPTION
**Pull request by ggalmazor**
(nulls and undefineds)

### CODE REVIEW

File: lib/reused/option.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+class Option {
+  static of(value) {
+    if (value instanceof Option)
+      return value;
+    if (value === null || value === undefined)
+      return none;
```
Comment (clint-tseng) [inline lib/reused/option.js]: I think this will fail the linter but I'll take a look at it.

Comment (clint-tseng) [reply lib/reused/option.js]: (use before declaration)

File: lib/reused/option.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  get() {
+    return this.value;
+  }
+
+  map(fn) {
+    return Option.of(fn(this.value));
+  }
+
+  filter(predicate) {
+    return predicate(this.value) ? this : none;
```
Comment (clint-tseng) [inline lib/reused/option.js]: Loose truthiness; intentional? (as opposed to `=== true`)

File: lib/reused/option.js (hunk 1 above)
Comment (ggalmazor) [reply lib/reused/option.js]: I run eslint with the current rules definition and it didn't complain... Maybe I've got the rules wrong...

File: lib/reused/option.js (hunk 2 above)
Comment (ggalmazor) [reply lib/reused/option.js]: Hmmm, I'm assuming that a predicate will always produce a boolean, but maybe you're right and we would benefit of a type and value check...

File: lib/reused/option.js (hunk 1 above)
Comment (clint-tseng) [reply lib/reused/option.js]: then i am wrong!

**Pull request merged**